functions/src/shared/s3-answers.ts

```diff
@@ -1,51 +1,104 @@
 const parquet = require('parquetjs');
 
-export interface AnswerData {
-  // leaving out everything but the resource_url and run_key which is what we care about
-  answer?: any;
+export interface AnswerMetadata {
   resource_url: string;
+  // logged-in user
+  platform_id?: string;
+  resource_link_id?: string;
+  platform_user_id?: string;
+  // anonymous user
   run_key?: string;
 }
 
+export interface AnswerData extends AnswerMetadata {
+  // leaving out everything but the resource_url and run_key which is what we care about
+  answer?: any;
+  report_state?: any;
+  version?: any;
+}
+
 export const schema = new parquet.ParquetSchema({
   submitted: { type: 'BOOLEAN', optional: true },
-  run_key: { type: 'UTF8' },
+  run_key: { type: 'UTF8', optional: true },
   platform_user_id: { type: 'UTF8', optional: true },
   id: { type: 'UTF8' },
   context_id: { type: 'UTF8', optional: true },
   class_info_url: { type: 'UTF8', optional: true },
   platform_id: { type: 'UTF8', optional: true },
   resource_link_id: { type: 'UTF8', optional: true },
-  type: { type: 'UTF8' },
+  type: { type: 'UTF8', optional: true },
   question_id: { type: 'UTF8' },
   source_key: { type: 'UTF8' },
-  question_type: { type: 'UTF8' },
-  tool_user_id: { type: 'UTF8' },
+  question_type: { type: 'UTF8', optional: true },
+  tool_user_id: { type: 'UTF8', optional: true },
   answer: { type: 'UTF8' },
   resource_url: { type: 'UTF8' },
```

Comment:
There will be some AP answer documents which have an incorrect resource_url. I fixed that here: https://github.com/concord-consortium/activity-player/pull/145 
I think we can just ignore those old answers for now, I think I got those fixed before the AP was used by real students. I'm just bring it up incase we can't find some data in the reports.